crates/verifier/src/lib.rs

```diff
@@ -103,16 +104,25 @@ impl Verifier<state::Initialized> {
         let mut ctx = mux_fut.poll_with(mt.new_context()).await?;
 
         // Receives protocol configuration from prover to perform compatibility check.
-        let protocol_config = mux_fut
-            .poll_with(async {
-                let peer_configuration: ProtocolConfig = ctx.io_mut().expect_next().await?;
-                self.config
-                    .protocol_config_validator()
-                    .validate(&peer_configuration)?;
-
-                Ok::<_, VerifierError>(peer_configuration)
-            })
-            .await?;
+        let protocol_config: ProtocolConfig = mux_fut.poll_with(ctx.io_mut().expect_next()).await?;
+
+        if let Err(e) = self
+            .config
+            .protocol_config_validator()
+            .validate(&protocol_config)
+        {
+            mux_fut
+                .poll_with(
+                    ctx.io_mut()
+                        .send::<Result<(), PeerError>>(Err(PeerError::from(&e))),
+                )
+                .await?;
+
+            mux_ctrl.close();
+            mux_fut.await?;
+
+            return Err(e.into());
+        }
```

Comment:
we also need to send `Ok(())` to prover if validation succeeds right? now when I run the example where validation passed, prover just hung there because it's waiting for the validation result